1570.py

```diff
@@ -0,0 +1,20 @@
+class SparseVector:
+    def __init__(self, nums: List[int]):
+        self.hashmap = {}
+        for i in range(len(nums)):
+            num = nums[i]
+            if num != 0:
+                self.hashmap[i] = num
+
+    # Return the dotProduct of two sparse vectors
+    def dotProduct(self, vec: 'SparseVector') -> int:
+        total = 0
+        for i, num in self.hashmap.items():
```

Comment:
The runtime of this function is 

O(a + b) where a = len(non zeroes in self), b = len(non zeroes in vec)

Can you optimize this code so that the runtime is O(min(a,b))